lnwire/onion_message.go

```diff
@@ -0,0 +1,79 @@
+package lnwire
+
+import (
+	"bytes"
+	"io"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+)
+
+// OnionMessage is a message that carries an onion-encrypted payload.
+// This is used for BOLT12 messages.
+type OnionMessage struct {
+	// BlindingPoint is the route blinding ephemeral pubkey to be used for
+	// the onion message.
+	BlindingPoint *btcec.PublicKey
+
+	// OnionBlob is the raw serialized mix header used to relay messages in
+	// a privacy-preserving manner. This blob should be handled in the same
+	// manner as onions used to route HTLCs, with the exception that it uses
+	// blinded routes by default.
+	OnionBlob []byte
+}
```

Comment:
No need for an `ExtraData` field here?

Reply:
not sure if we need it on this level tho, why would I use a blinded route and add more information basically unblinded, I think the bulk of extra data will be in `onionmsg_tlv` where we defintily need the ExtraData, but I mean there is no cost of provisioning it here as well.